### PR TITLE
feat(contributor): extend intro length

### LINKED ISSUE #393: Contributor intro is limited to 255 characters
image

Negative side effect :
- Some intros need more than 255 characters
- Including any type of svg path or other small visual code won't work.

### PR DESCRIPTION
**Pull request by gregoirelacoste**
closes #393

### CODE REVIEW

File: migrations/Version20210615164457.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * Auto-generated Migration: Please modify to your needs!
+ */
+final class Version20210615164457 extends AbstractMigration
+{
+    public function up(Schema $schema) : void
+    {
+        // this up() migration is auto-generated, please modify it to your needs
+        $this->abortIf('mysql' !== $this->connection->getDatabasePlatform()->getName(), 'Migration can only be executed safely on \'mysql\'.');
+
+        $this->addSql('ALTER TABLE contributor CHANGE intro intro LONGTEXT DEFAULT NULL');
```
Comment (lutangar) [inline migrations/Version20210615164457.php:18]: > A TEXT column with a maximum length of 4,294,967,295 or 4GB

Looks a bit extravagant compared to the previous `VARCHAR(255)`  :)

> https://stackoverflow.com/questions/28207647/how-to-get-doctrine-text-type

File: src/Entity/Contributor.php [hunk 2]
```diff
@@ -58,7 +58,6 @@ class Contributor implements ImageUploadable
      * @var string
      *
      * @ORM\Column(name="intro", type="string", length=255, nullable=true)
-     * @Assert\Length(max="255")
```
Comment (lutangar) [inline src/Entity/Contributor.php:61]: You could keep the length assertion but make it longer.

**Review by lutangar — APPROVED** [review]

**Comment by lutangar** [comment]
CS need to be fixed :
```
docker-compose exec php composer fix-cs
```

File: src/Entity/Contributor.php (hunk 2 above)
Comment (gregoirelacoste) [reply src/Entity/Contributor.php:61]: do you know what command to use to generate automatic migration ? migrations:diff generate something wrong

**Comment by gregoirelacoste** [comment]
@lutangar just put TEXT (length=65535)

File: src/Entity/Contributor.php (hunk 2 above)
Comment (lutangar) [reply src/Entity/Contributor.php:61]:
```
doctrine:migrations:diff
```

... is the right command, it generate as lot's of "noise"' -- which is not normal and should be fixed sometimes -- which you should strip down.

**Pull request merged**